### ISSUE TITLE
Entity not working on Roborock Q Revo

### ISSUE THREAD

**Issue by Jezza34000** [humbertogontijo/homeassistant-roborock#457]
## Describe the bug : 

List of entity not working properly on Roborock Q Revo :

sound_volume (slider) : Does not work i got message : Failed to call service number/set_value. required key not provided @ data['value']
dock_mop_wash_mode_interval : Does not work, stuck at "unknow"
dock_washing_mode : Does not work, stuck at "unknow"
dock_dust_collection_mode : Stay at 0 (This value does not change for Q REVO / P10 there is no dust collection mode)

## Debug log
Get_status from my Q Revo (when idle) :

```text
{'msg_ver': 2, 'msg_seq': 502, 'state': 8, 'battery': 100, 'clean_time': 1317, 'clean_area': 21392500, 'error_code': 0, 'map_present': 1, 'in_cleaning': 0, 'in_returning': 0, 'in_fresh_state': 1, 'lab_status': 1, 'water_box_status': 1, 'back_type': -1, 'wash_phase': 0, 'wash_ready': 1, 'wash_status': 0, 'fan_power': 104, 'dnd_enabled': 0, 'map_status': 3, 'is_locating': 0, 'lock_status': 0, 'water_box_mode': 203, 'distance_off': 125, 'water_box_carriage_status': 1, 'mop_forbidden_enable': 1, 'camera_status': 1, 'is_exploring': 0, 'adbumper_status': [0, 0, 0], 'water_shortage_status': 0, 'dock_type': 8, 'dust_collection_status': 0, 'auto_dust_collection': 1, 'avoid_count': 63, 'mop_mode': 300, 'debug_mode': 0, 'in_warmup': 0, 'collision_avoid_status': 1, 'switch_map_mode': 0, 'dock_error_status': 0, 'charge_status': 1, 'unsave_map_reason': 4, 'unsave_map_flag': 0, 'dry_status': 0, 'rdt': 0, 'clean_percent': 0, 'rss': 2, 'dss': 681, 'common_status': 0, 'last_clean_t': 1693480663, 'corner_clean_mode': 0}

```

**Comment by iddiek** [comment]
See #454 for a temporary fix

**Comment by Jezza34000** [comment]
I'm also investigating on my side in parallel, and I found some interesting stuff using the debug mode. 
I sent a pull request on python-roborock.
https://github.com/humbertogontijo/python-roborock/pull/114

**Comment by iddiek** [comment]
Doing the same, working also in docking commands.
Are you sure about mop mode?
mine command are different from yours
please take a look 
https://github.com/humbertogontijo/python-roborock/pull/115

**Comment by Lash-L** [comment]
Make sure you are both talking about the same thing. There is mop mode and mop intensity.

**Comment by Jezza34000** [comment]
Te be sure, when we talk about "mop_mode"  we re talking about this parameter ? @iddiek 

![Screenshot_2023-09-04-19-47-26-485_com roborock smart~2](https://github.com/humbertogontijo/homeassistant-roborock/assets/57314417/8e299076-bff3-46cd-99c3-b4cf56b0c4c8)

**Comment by iddiek** [comment]
Sorry for misunderstood, mop intensity is right, what about renaming in "route mode" :)

**Comment by Jezza34000** [comment]
OK, then :
Parameter "mop_mode" is "Route" in app
Parameter "water_box_mode" is "Water Flow" in app

That mean "mop_mode" code are exactly same as "RoborockMopModeS8ProUltra" and Q Revo dosen't need a new class for defining it level...

**Comment by iddiek** [comment]
But actually selecting standard doesn't do anithing. Also fast mode is
unavailable.

Il lun 4 set 2023, 20:07 Jezza ***@***.***> ha scritto:

**Comment by Lash-L** [comment]
> Sorry for misunderstood, mop intensity is right, what about renaming in "route mode" :)

Not all devices call it route is my understanding. I wasn't the one who added that parameter. But a comment would be a good move

**Comment by Jezza34000** [comment]
@iddiek you're right but this will fix the problem : 

```
@dataclass
class P10Status(Status):
    fan_power: Optional[RoborockFanSpeedP10] = None
    water_box_mode: Optional[RoborockMopIntensityV2] = None
    mop_mode: Optional[RoborockMopModeS8ProUltra] = None
```

**Comment by Jezza34000** [comment]
@iddiek you're right but this will fix the problem : 

```
@dataclass
class P10Status(Status):
    fan_power: Optional[RoborockFanSpeedP10] = None
    water_box_mode: Optional[RoborockMopIntensityV2] = None
    mop_mode: Optional[RoborockMopModeS8ProUltra] = None
```

**Comment by Jezza34000** [comment]
> dock_mop_wash_mode_interval : Does not work, stuck at "unknow"
> dock_washing_mode : Does not work, stuck at "unknow"

Fixed in new PR : https://github.com/humbertogontijo/python-roborock/pull/117

**Comment by Jezza34000** [comment]
Fixed with PR 
https://github.com/humbertogontijo/python-roborock/pull/114
https://github.com/humbertogontijo/python-roborock/pull/117

**Comment by broyuken** [comment]
What about the volume? I have that same issue on an S7
